src/main/cpp/DragonVision/DragonVision.cpp

```diff
@@ -174,33 +184,38 @@ std::optional<VisionData> DragonVision::GetVisionDataToNearestStageTag(VISION_EL
 
 std::optional<VisionData> DragonVision::GetVisionDataToNearestTag()
 {
-	DragonCamera *selectedCam = nullptr;
-	std::optional<int> launcherTagId = m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHER]->GetAprilTagID();
-	std::optional<int> placerTagId = m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::PLACER]->GetAprilTagID();
+	std::optional<VisionData> selectedData = std::nullopt;
+	std::optional<VisionData> launcherData = m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHER]->GetDataToNearestAprilTag();
```

Comment:
it may still be able for these cameras to be null. Not sure if we need a check before they are null and we call functions on them.